File: backend/src/main/java/kg/taskflow/service/AuthService.java
package kg.taskflow.service;

import kg.taskflow.dto.auth.AuthResponse;
import kg.taskflow.dto.auth.LoginRequest;
import kg.taskflow.dto.auth.RegisterRequest;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    AuthResponse refreshToken(String refreshToken);

    void logout(String refreshToken);

    void forgotPassword(String email);

    void resetPassword(String token, String newPassword);

    boolean verifyResetToken(String token);
}

File: frontend/src/services/auth.service.ts
import api from './api';
import { AuthResponse, LoginRequest, RegisterRequest } from '@/types/auth.types';

export const authService = {
  async login(data: LoginRequest): Promise<AuthResponse> {
    const response = await api.post<AuthResponse>('/auth/login', data);
    return response.data;
  },

  async register(data: RegisterRequest): Promise<AuthResponse> {
    const response = await api.post<AuthResponse>('/auth/register', data);
    return response.data;
  },

  async refreshToken(refreshToken: string): Promise<AuthResponse> {
    const response = await api.post<AuthResponse>('/auth/refresh', { refreshToken });
    return response.data;
  },

  async logout(refreshToken: string): Promise<void> {
    await api.post('/auth/logout', { refreshToken });
  },

  async forgotPassword(email: string): Promise<void> {
    await api.post('/auth/forgot-password', { email });
  },

  async resetPassword(token: string, password: string): Promise<void> {
    await api.post('/auth/reset-password', { token, password });
  },

  async verifyResetToken(token: string): Promise<{ valid: boolean }> {
    const response = await api.get<{ valid: boolean }>(`/auth/verify-reset-token/${token}`);
    return response.data;
  },
};
